src/v/kafka/server/group.h

```diff
@@ -615,7 +625,7 @@ class group {
     ctx_log _ctxlog;
     ctx_log _ctx_txlog;
 
-    mutex _tx_mutex;
+    absl::flat_hash_map<model::producer_id, ss::lw_shared_ptr<mutex>> _tx_locks;
```

Comment:
How is `_tx_locks` garbage collected?

Reply:
I added a commit with garbage collection